scylla/src/statement/prepared_statement.rs

```diff
@@ -237,13 +237,9 @@ impl PreparedStatement {
             return Ok(None);
         }
 
-        let partition_key =
-            self.extract_partition_key(serialized_values)
-                .map_err(|err| match err {
-                    PartitionKeyExtractionError::NoPkIndexValue(_, _) => {
-                        QueryError::ProtocolError("No pk indexes - can't calculate token")
-                    }
-                })?;
+        let partition_key = self
+            .extract_partition_key(serialized_values)
+            .map_err(|_| ProtocolError::PartitionKeyExtraction)?;
```

Comment:
That's risky that you no longer pattern match on the underlying error. This way, if someone extends `PartitionKeyExtractionError` with a new different variant, then there is a room for a bug - the new variant might be still converted to `ProtocolError::PartitionKeyExtraction` with a misleading description and the compiler won't complain.